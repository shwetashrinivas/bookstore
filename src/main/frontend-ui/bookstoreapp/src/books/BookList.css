.listbooks {
    display: flex;
    flex-wrap: wrap;
}

.books {
    margin-left: 5%;
}
  