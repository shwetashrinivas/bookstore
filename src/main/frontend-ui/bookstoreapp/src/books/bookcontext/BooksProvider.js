import React, { createContext, useState, useEffect } from "react";
import axios from "axios";

export const BooksContext = createContext();

const BooksProvider = ({ children }) => {
  const [books, setBooks] = useState({
    books: [],
  });

  useEffect(() => {
    const getBooks = async () => await axios.get("http://localhost:8090/books");
    getBooks()
      .then(({ data }) => {
        setBooks([...data]);
      })
      .catch(() => {});
  }, []);

  return (
    <BooksContext.Provider
      value={{
        setBooks,
        books,
      }}
    >
      {children}
    </BooksContext.Provider>
  );
};

export default BooksProvider;
