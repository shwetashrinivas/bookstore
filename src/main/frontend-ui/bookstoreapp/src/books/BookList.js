import React, { useContext } from "react";
import Book from "../common/book.js";
import "./BookList.css";
import { BooksContext } from "./bookcontext/BooksProvider.js";

export default function BookList() {
  const { books } = useContext(BooksContext);

  return (
    <div className="books">
      {books.length ? (
        <div data-testid="list-books" className="listbooks">
          {books.map((book) => {
            return <Book book={book} key={book?.id} />;
          })}
        </div>
      ) : (
        <p data-testid="no-book">No books available!</p>
      )}
    </div>
  );
}
