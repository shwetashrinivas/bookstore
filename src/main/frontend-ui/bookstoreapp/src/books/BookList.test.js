import { render, screen } from "@testing-library/react";
import BookList from "./BookList.js";
import { BooksContext } from "./bookcontext/BooksProvider.js";

const mockBookList = [
  {
    id: 1,
    title: "Tale of two cities",
    author: "Charles Dickens",
    price: 350.0,
    description:
      "The novel tells the story of the French Doctor Manette, his 18-year-long imprisonment in the Bastille in Paris, and his release to live in London with his daughter Lucie whom he had never met. The story is set against the conditions that led up to the French Revolution and the Reign of Terror.",
    coverImageUrl: "1586_f1601742134.png",
    rating: 5.0,
    quantity: 3,
  },
  {
    id: 2,
    title: "Little women",
    author: "Louisa May Alcott",
    price: 450.0,
    description:
      "It is set during and after the Civil War and tells the story of the March family, principally the four girls: Meg, Jo, Beth, and Amy. The girls struggle and grow as a family while their father is serving in the war and learn the value of hard work, self-sacrifice, and love.",
    coverImageUrl: "1586_f1601742134.png",
    rating: 5.0,
    quantity: 4,
  },
];

const renderBooks = () => {
  return (
    <BooksContext.Provider
      value={{
        books: mockBookList,
        setBooks: () => {},
      }}
    >
      <BookList />
    </BooksContext.Provider>
  );
};

describe("BookList", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should rendering book list component", () => {
    const { container } = render(renderBooks());
    expect(container).toMatchSnapshot();
  });

  it("should rendering book list component with list of books", () => {
    render(renderBooks());
    expect(screen.getByTestId("list-books")).toBeInTheDocument();
  });

  it("should check book list is empty", () => {
    render(
      <BooksContext.Provider
        value={{
          books: [],
          setBooks: () => {},
        }}
      >
        <BookList />
      </BooksContext.Provider>
    );
    expect(screen.getByTestId("no-book")).toBeInTheDocument();
    expect(screen.getByTestId("no-book")).toHaveTextContent(
      "No books available!"
    );
  });
});
