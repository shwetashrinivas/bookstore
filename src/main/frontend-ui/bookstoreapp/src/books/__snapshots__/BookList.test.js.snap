// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList should rendering book list component 1`] = `
<div>
  <div
    class="books"
  >
    <p>
      Total books: 
      2
    </p>
    <div
      class="listbooks"
      data-testid="list-books"
    >
      <div
        class="book-component"
        data-testid="book"
      >
        <img
          alt="abc"
          data-testid="coverImageUrl"
          height="240"
          src="https://res.cloudinary.com/www-thepencilapp-com/image/upload/q_auto,f_auto/books/covers/1586_f1601742134.png"
          width="180"
        />
        <p
          class="title"
          data-testid="title"
        >
          Tale of two cities
        </p>
        <p
          class="author"
          data-testid="author"
        >
          by $
          Charles Dickens
        </p>
        <p
          class="price"
          data-testid="price"
        >
          RS $
          350
        </p>
      </div>
      <div
        class="book-component"
        data-testid="book"
      >
        <img
          alt="abc"
          data-testid="coverImageUrl"
          height="240"
          src="https://res.cloudinary.com/www-thepencilapp-com/image/upload/q_auto,f_auto/books/covers/1586_f1601742134.png"
          width="180"
        />
        <p
          class="title"
          data-testid="title"
        >
          Little women
        </p>
        <p
          class="author"
          data-testid="author"
        >
          by $
          Louisa May Alcott
        </p>
        <p
          class="price"
          data-testid="price"
        >
          RS $
          450
        </p>
      </div>
    </div>
  </div>
</div>
`;
